src/main/java/seedu/bigpp/command/buildercommand/BuilderListComponentCommand.java

```diff
@@ -20,27 +35,127 @@ public BuilderListComponentCommand(String arguments) {
+        ComponentList<?> componentList = dataStorage.stringToComponentListMap.get(componentType);
+        ArrayList<String> flagDescriptionArray = new ArrayList<String>();
+        for (int i = 1; i < userInputStringArray.length; i++) {
+            String flagCommand = userInputStringArray[i].trim();
 
-        // throw exception if component type is not valid eg. "list jfk"
-        if (!dataStorage.stringToComponentListMap.containsKey(componentTypeString)) {
-            throw new BuilderIncorrectComponentException();
+            switch (flagCommand) {
+            case NAME_FLAG:
+                try {
+                    if (userInputStringArray.length < i + 2) {
+                        throw new BuilderMissingNameException();
+                    }
+                    String flagArgument = userInputStringArray[i + 1].trim();
+                    componentList = filterByName(componentList, flagArgument);
+                    flagDescriptionArray.add("name: " + flagArgument);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    throw new BuilderInvalidNumberFlags();
+                }
+                break;
+            case BRAND_FLAG:
+                try {
+                    String flagArgument = userInputStringArray[i + 1].trim();
+                    componentList = filterByBrand(componentList, flagArgument);
+                    flagDescriptionArray.add("brand: " + flagArgument);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    throw new BuilderInvalidNumberFlags();
+                }
+                break;
+            case PRICE_FLAG:
+                try {
+                    if (userInputStringArray.length < i + 4) {
+                        throw new BuilderPriceMissingArguments();
+                    }
+                    if (!(userInputStringArray[i + 1].equals("/from") && userInputStringArray[i + 3].equals("/to"))) {
+                        throw new BuilderPriceFlagException();
+                    }
+                    String priceFrom = userInputStringArray[i + 2];
+                    String priceTo = userInputStringArray[i + 4];
+                    if (priceFrom.matches(".*d.*") || priceTo.matches(".*d.*")) {
+                        throw new BuilderInvalidPriceType();
+                    }
+                    if (Float.parseFloat(priceFrom) > Float.parseFloat(priceTo)) {
+                        throw new BuilderPriceLogicException();
+                    }
+                    if (Float.parseFloat(priceFrom) <= 0 || Float.parseFloat(priceTo) <= 0) {
+                        throw new BuilderInvalidPriceNumber();
+                    }
+                    componentList = filterByPrice(componentList, priceFrom, priceTo);
+                    flagDescriptionArray.add("price: from " + priceFrom + " to " + priceTo);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    throw new BuilderInvalidNumberFlags();
```

Comment:
This might be BuilderInvalidFlagArguments instead? I think if a user inputs all the flags but without arguments this exception will happen

Reply:
yeap, changed